src/test/java/org/cactoos/proc/ForEachTest.java

```diff
@@ -60,4 +63,24 @@ void testProcIterable() throws Exception {
         ).affirm();
     }
 
+    @Test
+    @SuppressWarnings("unchecked")
+    void worksWithGenerics() throws Exception {
+        final List<? super Number> list = new LinkedList<>();
+        new ForEach<>(
+            (Proc<? super Number>) list::add
```

Comment:
@andreoss instead of casting, if you need to force the type of something as a `Proc`, use `new ProcOf<Number>(list::add)` (this is really just to avoid using casting, because if for any reason types change in `ForEach`, we will detect it at runtime instead of compile time, also we should be able to remove the suppressWarnings). But I'm surprised we need to explicit the type in any case :/